tracer/test/Datadog.Trace.Tests/Debugger/LiveDebuggerTests.cs

```diff
@@ -4,25 +4,50 @@
 // </copyright>
 
 using System;
-using System.Collections.Generic;
 using System.Threading.Tasks;
 using Datadog.Trace.Agent.DiscoveryService;
 using Datadog.Trace.Configuration;
 using Datadog.Trace.Debugger;
 using Datadog.Trace.Debugger.Configurations;
-using Datadog.Trace.Debugger.Configurations.Models;
-using Datadog.Trace.Debugger.Models;
 using Datadog.Trace.Debugger.ProbeStatuses;
 using Datadog.Trace.Debugger.Sink;
-using Datadog.Trace.Debugger.Sink.Models;
 using Datadog.Trace.RemoteConfigurationManagement;
-using FluentAssertions;
+using Moq;
 using Xunit;
 
 namespace Datadog.Trace.Tests.Debugger;
 
 public class LiveDebuggerTests
 {
+    private readonly Mock<IDiscoveryService> _discoveryService;
+    private readonly Mock<IRemoteConfigurationManager> _remoteConfigurationManager;
+    private readonly Mock<ILineProbeResolver> _lineProbeResolver;
+    private readonly Mock<IDebuggerSink> _debuggerSink;
+    private readonly Mock<IProbeStatusPoller> _probeStatusPoller;
+    private readonly ConfigurationUpdater _configurationUpdater;
+
+    public LiveDebuggerTests()
+    {
+        _discoveryService = new Mock<IDiscoveryService>();
+        _discoveryService
+           .Setup(service => service.SubscribeToChanges(It.IsAny<Action<AgentConfiguration>>()))
+           .Callback(
+                (Action<AgentConfiguration> callback) => callback(
+                    new AgentConfiguration(
+                        configurationEndpoint: "configurationEndpoint",
+                        debuggerEndpoint: "debuggerEndpoint",
+                        agentVersion: "agentVersion",
+                        statsEndpoint: "traceStatsEndpoint",
+                        dataStreamsMonitoringEndpoint: "dataStreamsMonitoringEndpoint",
+                        clientDropP0: false)));
+
```

Comment:
I'd make an alternative suggestion....

In [a recent PR](https://github.com/DataDog/dd-trace-dotnet/pull/3197), I introduced `DiscoveryServiceMock` for [testing with the discovery service](https://github.com/DataDog/dd-trace-dotnet/blob/c52124bb297e01cd167eefd307aced9d31644728/tracer/test/Datadog.Trace.Tests/Agent/DiscoveryServiceMock.cs).

This handles the fact that config can change after the initial result, and that consumers should handle it correctly. Rather than using `Moq` here, I think it would be preferable to use that, and test the various scenarios in the `LiveDebugger` (e.g. Unsupported -> Supported -> Unsupported)

Reply:
We can expose `NotifySubscribers` method on the interface, that way method can be mocked easily with `Moq`🤔

Reply:
IMO Moq is horrible once you actually want to have any interesting behaviour, I'd much prefer use something like [this implementation](https://github.com/DataDog/dd-trace-dotnet/blob/c52124bb297e01cd167eefd307aced9d31644728/tracer/test/Datadog.Trace.Tests/Agent/DiscoveryServiceMock.cs). And we don't really want to expose `NotifySubscribers` on `IDiscoveryService`.